src/api/app/models/user_ldap_strategy.rb

```diff
@@ -334,50 +334,52 @@ def self.initialize_ldap_con(user_name, password)
     return unless defined?(CONFIG['ldap_servers'])
     require 'ldap'
     ldap_servers = CONFIG['ldap_servers'].split(":")
-    count = 0
 
     max_ldap_attempts = CONFIG.has_key?('ldap_max_attempts') ? CONFIG['ldap_max_attempts'] : 10
-
-    while count < max_ldap_attempts
-      count += 1
+    # Do 10 attempts to connect to one of the configured LDAP servers. LDAP server
+    # to connect to is chosen randomly.
+    max_ldap_attempts.times do
       server = ldap_servers[rand(ldap_servers.length)]
       conn = try_ldap_con(server, user_name, password)
-      if !conn.nil? && conn.bound?
-        return conn
-      end
+
+      return conn if conn.try(:bound?)
```

Comment:
[nitpick] Why not to change the variable name to something more easy to understand? Something like **connection** 😄